src/components/node-graph/components/step/socket-type-badge.tsx

```diff
@@ -0,0 +1,34 @@
+import { FileIcon } from "lucide-react";
+
+import { StepSocket } from "@/api";
+import { Badge } from "@/components/ui/badge";
+import { IconDeviconPlainPython } from "@/components/ui/icon-devicon-plain-python";
+
+type OwnProps = {
+  socket: StepSocket;
+};
+
+const SocketTypeBadge: React.FC<OwnProps> = ({ socket }) => {
+  return (
+    <Badge className={"w-fit border-dashed border-blue-300 text-[0.5rem] leading-[0.75rem]"} variant={"outline"}>
+      {socket.data_type === "PythonObject" ? (
+        <>
+          <IconDeviconPlainPython className="mr-1" />
+          {socket.data_type_metadata && (
+            <>
+              - <code className="ml-1">{socket.data_type_metadata.name as string}</code>
+            </>
+          )}
+        </>
+      ) : socket.data_type === "UniconFile" ? (
+        <>
+          <FileIcon className="mr-1 h-2 w-2" /> - File
+        </>
+      ) : (
+        <>{socket.data_type}</>
```

Comment:
probably will not happen after this feature is merged in but good to check if `data_type` is empty

<img width="869" alt="image" src="https://github.com/user-attachments/assets/6a43d8b1-c9c4-421e-b4f5-650d7ddd6263" />

Reply:
maybe dynamic type inference?
<img width="840" alt="image" src="https://github.com/user-attachments/assets/067b2eab-54a8-44a5-b0e5-682cacb6e8ae" />
 
> this happens because of seed data without `data_type`, it gets populated only after focusing on the value (which triggers an update)